SpatialConvolution.lua

```diff
@@ -139,3 +139,16 @@ function SpatialConvolution:__tostring__()
    end
    return s .. ')'
 end
+
+function SpatialConvolution:clearState()
+   self.finput = self.output.new()
```

Comment:
on the same note as the one from @dominikgrewe , wouldn't it be better to do `self.finput:set()` in here ? One can for example share the internal buffers inside the model to reduce memory footprint, wouldn't it be desirable to be kept after `clearState` ?